src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -1687,6 +1690,37 @@ public String getVariableNamePattern() {
 					})
 					.serializer(new EnumSerializer<>(Gene.class)));
 		}
+		EnumUtils<EntityRegainHealthEvent.RegainReason> regainReasons = new EnumUtils<>(EntityRegainHealthEvent.RegainReason.class, "heal reasons");
+		Classes.registerClass(new ClassInfo<>(EntityRegainHealthEvent.RegainReason.class, "healreason")
+			.user("(regen|heal) (reason|cause)")
+			.name("Heal Reason")
+			.description("The heal reason in a heal event")
+			.usage(regainReasons.getAllNames())
+			.examples("")
+			.since("INSERT VERSION")
+			.parser(new Parser<EntityRegainHealthEvent.RegainReason>() {
+				@Override
+				@Nullable
+				public EntityRegainHealthEvent.RegainReason parse(String s, ParseContext parseContext) {
+					return regainReasons.parse(s);
+				}
+				
+				@Override
+				public String toString(EntityRegainHealthEvent.RegainReason o, int flags) {
+					return regainReasons.toString(o, flags);
+				}
+				
+				@Override
+				public String toVariableNameString(EntityRegainHealthEvent.RegainReason o) {
+					return "regainreason:" + o.name();
+				}
+				
+				@Override
+				public String getVariableNamePattern() {
+					return "regainreason:\\w+";
```

Comment:
Shouldn't this be just ``\\S+"?

Reply:
I mean both work,
\\S is everything that isn't a space, and \\w is any letter of the alphabet or number or _

Reply:
It would be better to follow the rest of the BukkitClasses registered and stick to ``\\S+``

Reply:
Okay